### PR TITLE
Update & Extend documentation

### PR DESCRIPTION
**Pull request by bossenti**
While working with the tableau server-client-python library I stumbled about some bugs in the documentation which I fixed within this PR.
Furthermore, I've created documentation for the method `add_to_schedule()` of the `schedules`-endpoint as this was missing so far.

Hopefully this PR fits so far, otherwise feel free to let me know.

Thanks for this great library, I really like the API of the client. It's fun to work with it :)

### CODE REVIEW

File: docs/api-ref.md
Comment (bcantoni) [inline docs/api-ref.md:577]: Could you make these same changes (rename file_path->file and add as_job) to the call above so they match:
`datasources.publish(datasource_item, file_path, mode, connection_credentials=None)`

File: docs/api-ref.md
Comment (bcantoni) [inline docs/api-ref.md:2737]: For all 3 of these schedule-->schedules changes, could you also make them in the preceding heading? (Ex: `#### schedule.create`)

File: docs/api-ref.md
Comment (bcantoni) [inline docs/api-ref.md:2704]: Same: should be DatasourceItem

File: docs/api-ref.md
Comment (bcantoni) [inline docs/api-ref.md:2691]: This should be DatasourceItem (lower case s).

**Review by bcantoni — CHANGES_REQUESTED** [review]
@bossenti thanks for the contribution here! I've added a few corrections to make but otherwise it's looking good.

File: docs/api-ref.md
Comment (kramantab) [inline docs/api-ref.md:2702]: Slight edit suggested:  The identifier (`luid`) of the schedule to add the data source or workbook to.  
@bcantoni let me know if "luid" here is incorrect. Also, in the description we say "Adds a `DatasourceItem` or a `WorkbookItem` to a schedule.", but we looks like you can also add a flow?

File: docs/api-ref.md
Comment (kramantab) [reply docs/api-ref.md:2704]: @bcantoni - more a question for you.  it says here that "passing multiple items will be deprecated."  So does this mean you can do that today and this functionality will be deprecated in the future?

Comment (bossenti) [reply docs/api-ref.md:2704]: As I could get I from the source code, you can currently pass a workbook and a datasource item at the same time with one method call, which leads to PendingDeprecationWarning. A FlowItem can only be passed individually.

File: docs/api-ref.md
Comment (bossenti) [reply docs/api-ref.md:2702]: I`ve added the `FlowItem` to the method description.
If using "luid" is appropriate, I'll adpat the parameter description as well.

File: docs/api-ref.md
Comment (bossenti) [reply docs/api-ref.md:2737]: Done

File: docs/api-ref.md
Comment (bossenti) [reply docs/api-ref.md]: Done

File: docs/api-ref.md
Comment (bossenti) [reply docs/api-ref.md:577]: Done

**Comment by bossenti** [comment]
Thanks @kramantab and @bcantoni for reviewing.
I've updated the PR according to four comments.

File: docs/api-ref.md
Comment (bcantoni) [reply docs/api-ref.md:2702]: Good catch adding FlowItem as well.

@kramantab I think to be consistent with the REST API documentation we can keep the more generic "id" usage here rather than "luid". That might be a future task if the REST API docs are moving that direction: updating TSC id vs luid usage.

Comment (bossenti) [reply docs/api-ref.md]: I've updated the description correspondingly.

File: docs/api-ref.md
Comment (bcantoni) [reply docs/api-ref.md]: @kramantab it does sound that way (meaning: might be deprecated in the future). I'm happy with the wording @bossenti put in here. It mentions the possible deprecation without being to specific about it.

**Review by bcantoni — APPROVED** [review]

**Comment by bcantoni** [comment]
All looks good to me. Passing it over to @kramantab for any additional feedback or changes.

@bossenti after @kramantab approves, you can merge. Please choose the "Squash and merge" option to combine all your commits into one.

**Review by kramantab — APPROVED** [review]
Looking good Thanks!

**Comment by bossenti** [comment]
@bcantoni, @kramantab, unfortunately I can't merge the pull request because I lack the appropriate permissions:  `Only those with [write access](https://docs.github.com/articles/what-are-the-different-access-permissions) to this repository can merge pull requests.`

**Comment by bcantoni** [comment]
Done - thanks @bossenti for the contribution!

**Pull request merged**
